knack/parser.py

```diff
@@ -112,20 +141,22 @@ def load_command_table(self, cmd_tbl):
                 else:
                     param = CLICommandParser._add_argument(command_parser, arg)
                 param.completer = arg.completer
-
+                param.deprecate_info = arg.deprecate_info
             command_parser.set_defaults(
                 func=metadata,
                 command=command_name,
                 _command_validator=command_validator,
                 _argument_validators=argument_validators,
                 _parser=command_parser)
 
-    def _get_subparser(self, path):
+    def _get_subparser(self, path, group_table=None):
         """For each part of the path, walk down the tree of
         subparsers, creating new ones if one doesn't already exist.
         """
+        group_table = group_table or {}
```

Comment:
`def _get_subparser(self, path, group_table={}):` would remove the need for this

Reply:
This is a big no-no in Python. 
http://effbot.org/zone/default-values.htm